pkg/restore/prioritize_group_version.go

```diff
@@ -0,0 +1,369 @@
+		context.Background(),
+		"enableapigroupversions",
+		metav1.GetOptions{},
+	)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil
+		}
+
+		return nil, errors.Wrap(err, "getting enableapigroupversions config map from velero namespace")
+	}
+
+	return cm, nil
+}
+
+func parseUserPriorities(prioritiesData string) map[string]metav1.APIGroup {
+	ups := make(map[string]metav1.APIGroup)
+
+	// The user priorities will be in a string of the form
+	// rockbands.music.example.io=v2beta1,v2beta2\n
+	// orchestras.music.example.io=v2,v3alpha1\n
+	// subscriptions.operators.coreos.com=v2,v1
+
+	lines := strings.Split(prioritiesData, "\n")
+
+	for _, l := range lines {
+		if !isUserPriorityValid(l) {
+			continue
+		}
+
+		rgvs := strings.SplitN(l, "=", 2)
+		rg := rgvs[0]   // rockbands.music.example.io
+		vers := rgvs[1] // v2beta1,v2beta2
+
+		vs := strings.Split(vers, ",")
+
+		ups[rg] = metav1.APIGroup{
+			Versions: versionsToGroupVersionForDiscovery(vs),
+		}
+	}
+
+	return ups
+}
+
+func isUserPriorityValid(line string) bool {
+	// Line must have one and only one equal sign
+	if strings.Count(line, "=") != 1 {
+		return false
+	}
+
+	// Line must contain at least one character before and after equal sign
+	pair := strings.Split(line, "=")
+	if len(pair[0]) < 1 || len(pair[1]) < 1 {
+		return false
+	}
+
+	// Line must not contain any spaces
+	if strings.Count(line, " ") > 0 {
+		return false
+	}
+
+	return true
+}
+
+// versionsToGroupVersionForDiscovery converts version strings into a Kubernetes format
+// for group versions.
+func versionsToGroupVersionForDiscovery(vs []string) []metav1.GroupVersionForDiscovery {
+	gvs := make([]metav1.GroupVersionForDiscovery, len(vs))
+
+	for i, v := range vs {
+		gvs[i] = metav1.GroupVersionForDiscovery{
+			Version: v,
+		}
+	}
+
+	return gvs
+}
+
+// findTargetGroup looks for a particular group in from a list of groups in the
+// target cluster.
+func findTargetGroup(tgvs []metav1.APIGroup, group string) metav1.APIGroup {
+	for _, tg := range tgvs {
+		if tg.Name == group {
+			return tg
+		}
+	}
+
+	return metav1.APIGroup{}
+}
+
+// findSupportedUserVersion finds the first user priority version that both source
+// and target support.
+func findSupportedUserVersion(ugvs, tgvs, sgvs []metav1.GroupVersionForDiscovery) string {
+	for _, ug := range ugvs {
+		if versionsContain(tgvs, ug.Version) && versionsContain(sgvs, ug.Version) {
+			return ug.Version
+		}
+	}
+
+	return ""
+}
+
+// versionsContain will check if a version can be found in a a slice of versions.
+func versionsContain(list []metav1.GroupVersionForDiscovery, version string) bool {
+	for _, v := range list {
+		if v.Version == version {
+			return true
+		}
+	}
+
+	return false
+}
+
+// latestCommon will find the versions that are in common between the source and
+// target clusters. It will return the latest of the common versions.
+func latestCommon(sgvs, tgvs []metav1.GroupVersionForDiscovery) string {
```

Comment:
The fact that alpha and beta versions will be chosen as latest in the overall functionality should be called out in the documentation.

Reply:
That function and associated unit test need to be removed since I don't use alphabetical order any more to sort the versions. I use Kubernetes priority versions now.